src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+      currentState = { ...currentState, ...extraData };
+    } else
+    if (type === 'removeProperties') {
+      currentState = Object.keys(currentState).reduce((newState, key) => {
+        if (!keysToRemove.includes(key)) {
+          newState[key] = currentState[key];
+        }
+
+        return newState;
+      }, {});
```

Comment:
let's make it simpler
```suggestion
     for (const value of actions.keysToRemove) {
        delete state[value];
      }
      break;
```